be/src/storage/column_aggregate_func.cpp

```diff
@@ -193,6 +195,49 @@ class ReplaceAggregator<ArrayColumn, ArrayState> final : public ValueColumnAggre
     bool need_deep_copy() const override { return true; }
 };
 
+template <>
+class ReplaceAggregator<MapColumn, ColumnRefState> final : public ValueColumnAggregator<MapColumn, ColumnRefState> {
+public:
+    void reset() override { this->data().reset(); }
+
+    void aggregate_impl(int row, const ColumnPtr& src) override {
+        this->data().column = src;
+        this->data().row = row;
+    }
+
+    void aggregate_batch_impl(int start, int end, const ColumnPtr& src) override { aggregate_impl(end - 1, src); }
+
+    void append_data(Column* agg) override {
+        auto* col = down_cast<MapColumn*>(agg);
+        DCHECK_NOTNULL(this->data().column);
+        col->append(*this->data().column, this->data().row, 1);
+    }
+
+    bool need_deep_copy() const override { return true; }
+};
+
+template <>
+class ReplaceAggregator<StructColumn, ColumnRefState> final
```

Comment:
duplicate implements with Map/ArrayColumn?

Reply:
yeah, almost the same